app/deliveryoptions/utils.py

```diff
@@ -0,0 +1,727 @@
+    "{DownloadText}": get_download_text,
+    "{DownloadUrl}": get_download_url,
+    "{FAType}": get_file_authority_type,
+    "{FoiUrl}": get_foi_url,
+    "{ImageLibraryUrl}": get_image_library_url,
+    "{ItemNumOfFilesAndSizeInMB}": get_item_num_of_files_and_size_in_MB,
+    "{KeepersGalleryUrl}": get_keepers_gallery_url,
+    "{KewBookingSystemUrl}": get_kew_booking_system_url,
+    "{MaxItems}": get_max_items,
+    "{OpenDateDesc}": get_open_date_desc,
+    "{OpeningTimesUrl}": get_opening_times_url,
+    "{OrderUrl}": get_order_url,
+    "{PaidSearchUrl}": get_paid_search_url,
+    "{Price}": get_price,
+    "{ReadersTicketUrl}": get_readers_ticket_url,
+    "{RecordCopyingUrl}": get_record_copying_url,
+    "{RecordInformationType}": get_record_information_type,
+    "{RecordOpeningDate}": get_record_opening_date,
+    "{RecordUrl}": get_record_url,
+    "{AllWebsiteUrls}": get_all_website_urls,
+    "{SubsWebsiteUrls}": get_subsequent_website_urls,
+    "{FirstWebsiteUrl}": get_first_website_url,
+    "{FirstWebsiteUrlFull}": get_first_website_url_full,
+    "{WebsiteUrlText}": get_website_url_text,
+    "{YourOrderLink}": get_your_order_link,
+}
+
+# Dictionary to serve as a cache for file contents, preventing redundant file reads
+file_cache = {}
+
+
+def read_delivery_options(file_path: str) -> List:
+    # Check if file content is already in the cache
+    if file_path not in file_cache:
+        # Open the file in read mode
+        with open(file_path, "r") as file:
+            # Load the JSON content from the file
+            file_content = json.load(file)
+
+            # Cache the file content for future use
+            file_cache[file_path] = file_content
+
+    # Return the file content either from the cache or newly loaded
+    return file_cache[file_path]
+
+
+@cache
+def get_dcs_prefixes() -> List:
+    dcs = settings.DELIVERY_OPTIONS_DCS_LIST
+    return dcs.split()
+
+
+def distressing_content_match(reference: str) -> bool:
+    dcs_prefixes = get_dcs_prefixes()
+
+    return list(filter(reference.startswith, dcs_prefixes)) != []
+
+
+def get_record(cache: Dict, record_id: int):
+    try:
+        return cache["deliveryOptions"]["option"][record_id]
+    except Exception:
+        return None
+
+
+def html_replacer(string: str, record: dict, surrogate_data: List) -> str:
+    subs = re.findall(r"{[A-Za-z]*}", string)
+
+    for s in subs:
+        try:
+            func = deliveryOptionsTags[s]
+
+            # If the tag doesn't have any data (can happen with surrogate links),
+            # rather than have a string with something missing, just return an
+            # empty string and it won't get displayed
+            if f := func(record, surrogate_data):
+                string = string.replace(s, f)
+        except Exception:
+            raise
+    return string
+
+
+def html_builder(
+    delivery_option_data: Union[List, str],
+    record_data: Dict,
+    surrogate_data: List = [],
```

Comment:
`record_data: Dict`, -> `record: Record`
`surrogate_data` -> `api_surrogate_data` to distinguish from the data from json `delivery_option_data`

Is the default [] necessary? Mutable objects shared across different calls cause unexpected behaviours.